project/geosource/models.py

```diff
@@ -63,6 +63,39 @@ def get_type_from_data(cls, data):
         return types.get(type(data), cls.Undefined)
 
 
+class SourceReporting(models.Model):
+    SUCCESS = "0"
+    ERROR = "1"
+    WARNING = "2"
+    SOURCE_STATUS = (
+        (SUCCESS, "Success"),
+        (ERROR, "Error"),
+        (WARNING, "Warning"),
+    )
+    status = models.CharField(choices=SOURCE_STATUS, max_length=10, null=True)
+    message = models.CharField(max_length=255, default="")
+    started = models.DateTimeField(null=True)
+    ended = models.DateTimeField(null=True)
+    added_lines = models.PositiveIntegerField(default=0)
+    deleted_lines = models.PositiveIntegerField(default=0)
+    modified_lines = models.PositiveIntegerField(default=0)
+    errors = models.JSONField(default=list)
+
+    def save(self, *args, **kwargs):
+        self.ended = timezone.now()
```

Comment:
I think that is task report method that should update this field

Reply:
Are you talking about the methods in `project/geosource.task.py` ?

Reply:
Done !